File: Lecture 3/StarterCode - Java/app/src/main/java/com/inovaeg/androidtrainingpharos2020/finaldemo/NetflixLoginActivity.java
package com.inovaeg.androidtrainingpharos2020.finaldemo;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.inovaeg.androidtrainingpharos2020.R;

public class NetflixLoginActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_netflix_login);

        //TODO add text watcher to emailEditText to make button enabled when it's not empty

        //TODO add text watcher to passwordEditText to make button enabled when it's not empty

        //TODO make sure that both watchers check on the other edittext that it's not empty too

        //TODO add clickListener to login button to go to NetflixMovieOverviewActivity

    }
}

File: Lecture 3/StarterCode - Kotlin/app/src/main/java/com/inovaeg/androidtrainingpharos2020/finaldemo/NetflixLoginActivity.kt
package com.inovaeg.androidtrainingpharos2020.finaldemo

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import androidx.appcompat.app.AppCompatActivity
import com.inovaeg.androidtrainingpharos2020.R
import kotlinx.android.synthetic.main.activity_netflix_login.*

class NetflixLoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_netflix_login)

        email_edit_text.addTextChangedListener(object : TextWatcher{
            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
            override fun afterTextChanged(p0: Editable?) {
                login_btn.isEnabled = p0.toString().isNotEmpty() && password_edit_text.text.toString().isNotEmpty()
            }
        })

        password_edit_text.addTextChangedListener(object : TextWatcher{
            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
            override fun afterTextChanged(p0: Editable?) {
                login_btn.isEnabled = p0.toString().isNotEmpty() && login_btn.text.toString().isNotEmpty()
            }
        })

        login_btn.setOnClickListener {
            startActivity(Intent(this@NetflixLoginActivity, NetflixMovieOverviewActivity::class.java))
        }


    }
}